x/oracle/keeper/prophecy.go

```diff
@@ -140,14 +142,18 @@ func (k Keeper) AppendSignature(ctx sdk.Context, prophecyID []byte, ethereumAddr
 
 // CleanUpProphecy clean up outdated prophecy
 func (k Keeper) CleanUpProphecy(ctx sdk.Context) {
+	// it is low efficient to check outdated prophecy each block
+	if k.currentHeight%CleanUpFrequency != 0 {
```

Comment:
Please put spaces between your operators and variables; I assume this is modulus operator. This just reduces the frequency this gets called correct? Cant this result in going very long times without a cleanup if the function calling this is always a block offset from 1000?

Reply:
Yes, we need avoiding the keeper to check all prophecies in every block. Since the ProphecyLiftTime = 520000, 1000 just a small number compared with the life time.

Reply:
No, it actually works this way. If we have a prophecy for cross chain transfer in block X. Then from X + 52000, we may remove it since its ProphecyLiftTime. but it must be removed before X + 52000 + 1000. Which exact block depends on the last four digit of X.